app/controllers/listings_controller.rb

```diff
@@ -22,11 +33,34 @@ def create
   end
 
   def index
+    require_user
```

Comment:
this should probably be removed from the method and made a before action
before_action :require_user, only: [:index, :show]